pkg/reconciler/app/reconciler.go

```diff
@@ -594,9 +598,59 @@ func (r *Reconciler) ApplyChanges(ctx context.Context, app *v1alpha1.App) error
 		app.Status.PropagateInstanceStatus(instanceStatus)
 	}
 
+	// Sync start commands, populate container and buildpack start commands in app status.
+	{
+		logger.Debug("reconciling start commands")
+
+		startCommands := app.Status.StartCommands
+
+		containerConfig, err := r.fetchContainerCommand(app)
+		if err != nil {
+			return err
```

Comment:
It might be worth reflecting this error onto the app status instead so the developer can understand why the pull failed.